texar/torch/modules/connectors/connectors.py

```diff
@@ -144,19 +140,14 @@ def _mlp_transform(inputs: TensorStruct,
     else:
         size_list = flat_output_size
 
-    fc_output = concat_input
     if linear_layer is not None:
-        fc_output = linear_layer(fc_output)
+        fc_output = linear_layer(concat_input)
     if activation_fn is not None:
         fc_output = activation_fn(fc_output)
-
-    flat_output = torch.split(fc_output, size_list, dim=1)
+    elif linear_layer is None and activation_fn is None:
+        fc_output = concat_input
+    flat_output = split(fc_output, size_list, dim=1)
```

Comment:
Don't import individual functions from `torch`. Just call `torch.split`.

Reply:
And this part was changed by me, please revert it back. Your implementation here has problems: what if `linear_layer` is `None` but `activation_fn` is not?